File: WebApi/api/api/Models/CatalogMast.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace api.Models
{
    public class CATALOG_MAST
    {
        [Key]
        public long catalog_id { get; set; }

        [StringLength(15)]
        public string pdbrnd_code { get; set; }

        [StringLength(15)]
        public string pddsgn_code { get; set; }

        [StringLength(100)]
        public string dsgn_name { get; set; }

        [StringLength(400)]
        public string dsgn_desc { get; set; }

        [StringLength(200)]
        public string pic_file_path { get; set; }

        [StringLength(int.MaxValue)]
        public string pic_base64 { get; set; }

        [StringLength(10)]
        public string status { get; set; }

        [StringLength(15)]
        public string created_by { get; set; }

        public DateTime created_at { get; set; }

        [StringLength(15)]
        public string updated_by { get; set; }

        public DateTime updated_at { get; set; }

        //public virtual List<CatalogColor> catalogColorList { get; set; }

    }
}

File: web/src/app/_model/catalog-mast.ts
import { AppSetting } from '../_constants/app-setting';

export class CatalogMastView 
{
    public catalog_id: number = undefined;
    public pdbrnd_code: string = "";
    public pddsgn_code: string = "";
    public dsgn_name: string = "";
    public dsgn_desc: string ="";
    public pic_file_path : any = null;
    public pic_base64: string = "";
    public status: string = "";
    public created_by: string = "";
    public created_at: any = null;
    public updated_by: string = "";
    public updated_at :any= null;
    public catalogColors: DesignColorView[] = [];
}

export class DesignColorView
{
    public pdcolor_code:string ="";
    public pdcolor_name:string ="";
    public pic_file_path:string ="";
    public pic_base64:string ="";
    public catalog_file_path:string = "";
    public fullPath: string = "";

    
}

export class CatalogMastSearchView
{
    public pageIndex: number = 1;
    public itemPerPage: number = AppSetting.itemPerPage;
    public pddsgn_code: string ="";
    public dsgn_name: string ="";
    public dsgn_desc: string ="";
}